store/views.py

```diff
@@ -24,49 +31,89 @@ def bookDetailView(request, bid):
 
 @csrf_exempt
 def bookListView(request):
+    
     template_name = 'store/book_list.html'
+    get_data = request.GET
+    print(type(get_data))
+    print(get_data)
+    # titles=get_data.get('title')
+    # # if(titles==None):
+    # #     titles=''
+   
+    books=(Book.objects.filter(
+        title__icontains=get_data.get('title',''),
+        author__icontains=get_data.get('author',''),
+        genre__icontains=get_data.get('genre','')
+    ))
+
     context = {
-        'books': None, # set this to the list of required books upon filtering using the GET parameters
+        'books': books, # set this to the list of required books upon filtering using the GET parameters
                        # (i.e. the book search feature will also be implemented in this view)
     }
-    get_data = request.GET
+    
     # START YOUR CODE HERE
     
     
     return render(request, template_name, context=context)
 
 @login_required
 def viewLoanedBooks(request):
+    current_user=request.user
+    # print(current_user)
+    books=BookCopy.objects.filter(borrower=current_user)
+    # print(books)
+
+    
     template_name = 'store/loaned_books.html'
+
+
     context = {
-        'books': None,
+        'books': books,
     }
     '''
     The above key 'books' in the context dictionary should contain a list of instances of the 
     BookCopy model. Only those book copies should be included which have been loaned by the user.
     '''
     # START YOUR CODE HERE
+
     
 
 
     return render(request, template_name, context=context)
 
 @csrf_exempt
-@login_required
 def loanBookView(request):
-    response_data = {
-        'message': None,
-    }
+    alpha=request.user.is_authenticated
+    if not alpha:
+        return JsonResponse({"message":"login plz"})
+    book_id=request.POST['bid']
```

Comment:
You are directly accessing POST data without checking if it even exists. This may lead to server crash if a user access this endpoint with invalid request data. The good behavior would have been to throw a client error (400), rather than server error (500).